src/utils/ldap.py

```diff
@@ -242,8 +242,38 @@ def update(self, rdn, attr_dict, new_rdn=False):
 
         """
         dn = rdn + "," + self.basedn
-        actual_entry = self.search(rdn, attrs=None)
-        ldif = modlist.modifyModlist(actual_entry[0], attr_dict, ignore_oldexistent=1)
+        current_entry = self.search(rdn, attrs=None)
+        # Previously, we used modifyModlist, which directly uses the lib system libldap
```

Comment:
Wouldn't it be more readable to split it into a function, named like `ldif_modifyModlist_finegrained` ?